src/Surfnet/StepupMiddleware/MiddlewareBundle/Console/Command/ReplayEventsCommand.php

```diff
@@ -0,0 +1,113 @@
+{
+    protected function configure()
+    {
+        $this
+            ->setName('middleware:event:replay')
+            ->setDescription('wipes all read models and repopulates the tables from the event store')
+            ->addOption(
+                'increments',
+                'i',
+                InputOption::VALUE_REQUIRED,
+                'The amount of events that are replayed at once (repeated until all events are replayed)',
+                1000
+            );
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        /** @var KernelInterface $kernel */
+        $kernel      = $this->getApplication()->getKernel();
+        $environment = $kernel->getEnvironment();
+        /** @var FormatterHelper $formatter */
+        $formatter = $this->getHelper('formatter');
+
+        if (!in_array($environment, ['dev_event_replay', 'prod_event_replay'])) {
+            $output->writeln($formatter->formatBlock(
+                ['', 'This command may only be executed using env "dev_event_replay" or "prod_event_replay"', ''],
+                'error'
+            ));
+
+            return;
+        }
+
+        /** @var QuestionHelper $interrogator */
+        $interrogator = $this->getHelper('question');
+        if ($environment === 'prod_event_replay') {
+            $wantToRunOnProd = new ConfirmationQuestion(
+                '<question>You have selected to run this on production. Have you disabled all access to the production '
+                . 'environment? (y/N)</question>',
+                false
+            );
+
+            if (!$interrogator->ask($input, $output, $wantToRunOnProd)) {
+                $output->writeln('<comment>Not starting the replay</comment>');
+                return;
+            }
+        }
+
+        $increments = (int) $input->getOption('increments');
+        if ($increments < 1) {
+            $output->writeln($formatter->formatBlock(
+                sprintf('Increments must be a positive integer, "%s" given', $input->getOption('increments')),
+                'error'
+            ));
+            return;
+        }
+
+        $output->writeln(['', $formatter->formatBlock(['', 'WARNING!!!!', ''], 'error'), '']);
+
+        $question = <<<QUESTION
+You are about to WIPE all read data and recreate all data based on the events present, in steps of %d.
+
+  <%s>>> This can take a while and may NOT be interrupted <<</%s>
+
+Are you sure you wish to continue? (y/N)
+QUESTION;
+
+        $question = sprintf($question, $increments, 'error', 'error');
+        $areYouSure = new ConfirmationQuestion(sprintf("<question>%s</question>\n", $question), false);
+        if (!$interrogator->ask($input, $output, $areYouSure)) {
+            $output->writeln('<comment>Replay cancelled!</comment>');
+
+            return;
+        }
+
+        $output->writeln(['', '<info>Starting Event Replay</info> <comment>DO NOT INTERRUPT THIS!</comment>', '', '']);
```

Comment:
All these warnings suggest that, when interrupted, the system is left in an unrecoverable state. Perhaps merely suggest once not to interrupt the process and restart it if interrupted.